### PR TITLE
feat: website route (downloads, help, plugins, showcase) translation

### PR DESCRIPTION
**Pull request by barchielAbyss**
🥳 this is last hard-coded in customized react template, appreciate everyone

Changes:

added translation :
```
   zh/showcase
   zh/help
   zh/downloads
   zh/plugins
```

change json key, make sure the translation json order is the same as the file directory order
```
{
    hero.webpage  ->   hero.component
    arrowAnim.webpage  ->  arrowAnim.component
}
```

### CODE REVIEW

**Comment by barchielAbyss** [comment]
@SkyeYoung 
hi, if u have a litte times, please check this issue, i want add A new component 2 fix this. in before time, i'm written Vue component [Affix](https://fe.xiaoman.cn/amumu/components/affix.html), i want write React version component for after time.

https://github.com/apache/apisix-website/issues/1107

File: website/i18n/zh/code.json [hunk 1]
```diff
@@ -35,6 +27,86 @@
...
+  "download.website.history.incubator": {
+    "message": " 存储着 Apache APISIX 旧版孵化器项目的备份归档，你也可以随时查阅它们。",
+    "description": "&nbsp;hosts older releases when APISIX was an incubator project."
+  },
+  "download.website.verify.title": {
+    "message": "验证 Apache APISIX 版本号",
+    "description": "Verify the releases"
+  },
+  "download.website.link.PGP_KEY": {
+    "message": "获取PGP签名密钥",
```
Comment (guitu168) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "获取 PGP 签名密钥",
```

File: website/i18n/zh/code.json [hunk 2]
```diff
@@ -35,6 +27,86 @@
...
+  "download.website.link.incubator": {
+    "message": "Incubating Archive repository",
+    "description": "Incubating Archive repository"
+  },
+  "download.website.history.incubator": {
+    "message": " 存储着 Apache APISIX 旧版孵化器项目的备份归档，你也可以随时查阅它们。",
+    "description": "&nbsp;hosts older releases when APISIX was an incubator project."
+  },
+  "download.website.verify.title": {
+    "message": "验证 Apache APISIX 版本号",
```
Comment (hf400159) [inline website/i18n/zh/code.json:59]:
```suggestion
    "message": "验证 Apache APISIX 版本",
```

File: website/i18n/zh/code.json [hunk 3]
```diff
@@ -35,6 +27,86 @@
...
+    "description": "Verify the releases"
+  },
+  "download.website.link.PGP_KEY": {
+    "message": "获取PGP签名密钥",
+    "description": "Get PGP signatures KEYS"
+  },
+  "download.website.verify.step1": {
+    "message": "你可以使用 PGP 或 SHA 签名验证 Apache APISIX 的文件完整性，使用 GPG 或 PGP 验证 PGP 签名的合法性， 建议从主目录获取密钥文件和ASC签名文件。",
```
Comment (guitu168) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "你可以使用 PGP 或 SHA 签名验证 Apache APISIX 的文件完整性，使用 GPG 或 PGP 验证 PGP 签名的合法性，建议从主目录获取密钥文件和 ASC 签名文件。",
```

File: website/i18n/zh/code.json [hunk 4]
```diff
@@ -35,6 +27,86 @@
...
+    "description": "It is essential that you verify the integrity of the downloaded files using the PGP or SHA signatures. The PGP signatures can be verified using GPG or PGP. Please download the KEYS as well as the asc signature files for relevant distribution. It is recommended to get these files from the main distribution directory and not from the mirrors."
+  },
+  "download.website.verify.step2": {
+    "message": "如果需要验证二进制文件/源代码，您可以从主目录下载相关的 asc 文件，并遵循下面的指南。",
```
Comment (guitu168) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "如果需要验证二进制文件/源代码，您可以从主目录下载相关的 ACS 文件，并遵循下面的指南。",
```

**Review by guitu168 — COMMENTED** [review]
Can you fix the formatting problem first? 😆

**Comment by barchielAbyss** [comment]
> Can you fix the formatting problem first? 😆

Oh, i'm sorry, because I set up my Vscode editor config: 
``` json
{
  "editor.formatOnSave": true,
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[typescriptreact]": {
    "editor.defaultFormatter": "vscode.typescript-language-features"
  },
  "[json]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  }
}
```
what formatting tool do you use, can you recommend me?

**Comment by guitu168** [comment]
> what formatting tool do you use, can you recommend me?
cc @barchielAbyss  ohh, for Chinese format detection in JSON files, I don't have a tool recommendation.

File: website/i18n/zh/code.json [hunk 5]
```diff
@@ -3,29 +3,21 @@
...
-  },
-  "hero.webpage.title.fragment2": {
-    "message": "API 流量",
-    "description": "API Traffic"
-  },
-  "hero.webpage.title.fragment3": {
-    "message": "管理组件",
-    "description": "management"
+  "showcase.website.title": {
+    "message": "产品展示",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "用户列表",
```

File: website/i18n/zh/code.json [hunk 6]
```diff
@@ -3,29 +3,21 @@
...
+    "description": "Showcase"
   },
-  "hero.webpage.subtitle.content": {
-    "message": "Apache APISIX 是一个动态、实时、高性能的云原生 API 网关，提供了负载均衡、动态上游、灰度发布、服务熔断、身份认证、可观测性等丰富的流量管理功能。",
-    "description": "Apache APISIX provides rich traffic management features like Load Balancing, Dynamic Upstream, Canary Release, Circuit Breaking, Authentication, Observability, and more..."
+  "showcase.website.tips.used": {
+    "message": "他们都在使用 Apache APISIX 进行生产和研究",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "Apache APISIX 被众多产品与用户所信赖",
```

File: website/i18n/zh/code.json [hunk 7]
```diff
@@ -3,29 +3,21 @@
...
+    "description": "This project is used by all these folks"
   },
-  "hero.webpage.download.btn": {
-    "message": "立即体验",
-    "description": "Download"
+  "showcase.website.tips.wantUse": {
+    "message": "你也在使用它吗？告诉我们是如何使用，我们将会积极的收集和采纳你的意见。",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "你也在使用它吗？请让我们知道！",
```

File: website/i18n/zh/code.json [hunk 8]
```diff
@@ -35,6 +27,86 @@
     "message": "让产品与技术背后的细节更加可视化",
     "description": "We love open source."
   },
+  "download.website.title": {
+    "message": "开启智能网关之旅",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "立即下载",
```

**Comment by barchielAbyss** [comment]
> > what formatting tool do you use, can you recommend me?
> > cc @barchielAbyss  ohh, for Chinese format detection in JSON files, I don't have a tool recommendation.

What you want to say should be: *this commit does two things, which is not good, they should be done separately*
🤔

File: website/i18n/zh/code.json [hunk 9]
```diff
@@ -35,6 +27,86 @@
...
+    "description": "Downloads"
+  },
+  "download.website.subtitle": {
+    "message": "We love open source.",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "We love Open Source.",
```

File: website/i18n/zh/code.json [hunk 10]
```diff
@@ -35,6 +27,86 @@
...
+  "download.website.history.title": {
+    "message": "历史版本",
+    "description": "History Versions"
+  },
+  "download.website.history.all": {
+    "message": "如果你想要预览 Apache APISIX 已发布的所有版本，请查看 ",
+    "description": "Find all APISIX releases in the&nbsp;"
+  },
+  "download.website.link.archive": {
+    "message": "Archive repository",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json:47]:
```suggestion
    "message": "历史版本归档",
```

File: website/i18n/zh/code.json [hunk 11]
```diff
@@ -35,6 +27,86 @@
...
+    "description": "Archive repository"
+  },
+  "download.website.link.incubator": {
+    "message": "Incubating Archive repository",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json:51]:
```suggestion
    "message": "孵化时的历史版本",
```

File: website/i18n/zh/code.json [hunk 12]
```diff
@@ -35,6 +27,86 @@
...
+  "download.website.verify.step1": {
+    "message": "你可以使用 PGP 或 SHA 签名验证 Apache APISIX 的文件完整性，使用 GPG 或 PGP 验证 PGP 签名的合法性， 建议从主目录获取密钥文件和ASC签名文件。",
+    "description": "It is essential that you verify the integrity of the downloaded files using the PGP or SHA signatures. The PGP signatures can be verified using GPG or PGP. Please download the KEYS as well as the asc signature files for relevant distribution. It is recommended to get these files from the main distribution directory and not from the mirrors."
+  },
+  "download.website.verify.step2": {
+    "message": "如果需要验证二进制文件/源代码，您可以从主目录下载相关的 asc 文件，并遵循下面的指南。",
+    "description": "To verify the binaries/sources you can download the relevant asc files for it from main distribution directory and follow the below guide."
+  },
+  "help.website.title": {
+    "message": "需要帮助吗？",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "获取帮助",
```

File: website/i18n/zh/code.json [hunk 13]
```diff
@@ -35,6 +27,86 @@
...
+    "description": "NEED HELP?"
+  },
+  "help.website.subtitle": {
+    "message": "维护 Apache APISIX 的专业团队人员随时可以为你答疑解惑",
```
Comment (juzhiyuan) [inline website/i18n/zh/code.json]:
```suggestion
    "message": "Apache APISIX 维护者们将积极为你答疑解惑",
```

File: website/i18n/zh/code.json (hunk 11 above)
Comment (barchielAbyss) [reply website/i18n/zh/code.json:51]: I think the link of the **file repository** should not be translated directly. 
The file repository **path** and **title** corresponds to English, which may cause trouble for developers after translation.

Comment (juzhiyuan) [reply website/i18n/zh/code.json:51]: OK

File: website/i18n/zh/code.json (hunk 10 above)
Comment (barchielAbyss) [reply website/i18n/zh/code.json:47]: > 

I think the link of the **file repository** should not be translated directly. 
The file repository **path** or **name** corresponds to English, which may cause trouble for developers after translation.

File: website/i18n/zh/code.json (hunk 11 above)
Comment (barchielAbyss) [reply website/i18n/zh/code.json:51]: > 

like `line 47`

File: website/i18n/zh/code.json (hunk 10 above)
Comment (juzhiyuan) [reply website/i18n/zh/code.json:47]: OK

**Review by juzhiyuan — DISMISSED** [review]
LGTM

**Review by SkyeYoung — DISMISSED** [review]

**Comment by SkyeYoung** [comment]
> > Can you fix the formatting problem first? 😆
> 
> Oh, i'm sorry, because I set up my Vscode editor config:
> 
> ```json
> {
>   "editor.formatOnSave": true,
>   "[javascript]": {
>     "editor.defaultFormatter": "esbenp.prettier-vscode"
>   },
>   "[typescriptreact]": {
>     "editor.defaultFormatter": "vscode.typescript-language-features"
>   },
>   "[json]": {
>     "editor.defaultFormatter": "esbenp.prettier-vscode"
>   }
> }
> ```
> 
> what formatting tool do you use, can you recommend me?

We are using eslint, although it is not strictly a formatting tool. For some reason I don't use strict configuration, so it has a lot of problems. 

If you follow the README I wrote and install the appropriate dependencies, eslint should help you with some of the formatting issues by husky calling it before you commit. 

If there are some deficiencies or problems with our configuration, you are welcome to help us improve it.

**Review by guitu168 — DISMISSED** [review]

**Comment by SkyeYoung** [comment]
> What you want to say should be: _this commit does two things, which is not good, they should be done separately_

I'm working on a solution regarding the formatting of mixed Chinese and English, but I haven't had time to test it and actually use it for this project yet.

As a temporary solution, you can use zhlint.

**Review by juzhiyuan — APPROVED** [review]

**Review by yzeng25 — APPROVED** [review]

**Review by SkyeYoung — APPROVED** [review]

**Comment by lijing-21** [comment]
Hi @barchielAbyss  , thanks for the contribution!

Here is the Contributor T-shirt form[1], if you're interested, kindly take a look :)

[1] https://github.com/apache/apisix/blob/master/CONTRIBUTING.md#contributor-t-shirt

**Pull request merged**